internal/service/timestreaminfluxdb/db_instance.go

```diff
@@ -0,0 +1,1182 @@
+	r.SetDefaultDeleteTimeout(30 * time.Minute)
+
+	return r, nil
+}
+
+const (
+	// If not provided, CreateDbInstance will use the below default values
+	// for bucket and organization. These values need to be set in Terraform
+	// because GetDbInstance won't return them.
+	DefaultBucketValue       = "bucket"
+	DefaultOrganizationValue = "organization"
+	ResNameDbInstance        = "Db Instance"
+)
+
+type resourceDbInstance struct {
+	framework.ResourceWithConfigure
+	framework.WithTimeouts
+}
+
+func (r *resourceDbInstance) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = "aws_timestreaminfluxdb_db_instance"
+}
+
+// TIP: ==== SCHEMA ====
+// In the schema, add each of the attributes in snake case (e.g.,
+// delete_automated_backups).
+//
+// Formatting rules:
+// * Alphabetize attributes to make them easier to find.
+// * Do not add a blank line between attributes.
+//
+// Attribute basics:
+//   - If a user can provide a value ("configure a value") for an
+//     attribute (e.g., instances = 5), we call the attribute an
+//     "argument."
+//   - You change the way users interact with attributes using:
+//   - Required
+//   - Optional
+//   - Computed
+//   - There are only four valid combinations:
+//
+// 1. Required only - the user must provide a value
+// Required: true,
+//
+//  2. Optional only - the user can configure or omit a value; do not
+//     use Default or DefaultFunc
+//
+// Optional: true,
+//
+//  3. Computed only - the provider can provide a value but the user
+//     cannot, i.e., read-only
+//
+// Computed: true,
+//
+//  4. Optional AND Computed - the provider or user can provide a value;
+//     use this combination if you are using Default
+//
+// Optional: true,
+// Computed: true,
+//
+// You will typically find arguments in the input struct
+// (e.g., CreateDBInstanceInput) for the create operation. Sometimes
+// they are only in the input struct (e.g., ModifyDBInstanceInput) for
+// the modify operation.
+//
+// For more about schema options, visit
+// https://developer.hashicorp.com/terraform/plugin/framework/handling-data/schemas?page=schemas
+func (r *resourceDbInstance) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"allocated_storage": schema.Int64Attribute{
+				Required: true,
+				PlanModifiers: []planmodifier.Int64{
+					int64planmodifier.RequiresReplace(),
+				},
+				Validators: []validator.Int64{
+					int64validator.AtLeast(20),
+					int64validator.AtMost(16384),
+				},
+				Description: `The amount of storage to allocate for your DB storage type in GiB (gibibytes). 
+					This argument is required. This argument has a minimum value of 
+					20 and a maximum value of 16384`,
+			},
+			"arn": framework.ARNAttributeComputedOnly(),
+			"availability_zone": schema.StringAttribute{
```

Comment:
How does this get computed? Users can specify `Multi-AZ` or `Single-AZ` through the UI when creating an instance.

Reply:
Users can't provide an availability zone as an argument to `CreateDbInstance`. The availability zone is returned from `CreateDbInstanceOutput`. Users can use the argument `deployment_type` and choose either `SINGLE_AZ` or `WITH_MULTIAZ_STANDBY`, defaulting to `SINGLE_AZ`. This is considered different from an availability zone.